pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java

```diff
@@ -2477,9 +2530,18 @@ protected void handleEndTxnOnPartition(CommandEndTxnOnPartition command) {
         CompletableFuture<Optional<Topic>> topicFuture = service.getTopicIfExists(TopicName.get(topic).toString());
         topicFuture.thenAccept(optionalTopic -> {
             if (optionalTopic.isPresent()) {
-                optionalTopic.get().endTxn(txnID, txnAction, lowWaterMark)
+                // we only accept super user becase this endpoint is reserved for broker to broker communication
+                verifyTxnOwnershipForBrokerToBrokerCommands()
+                        .thenCompose(isOwner -> {
+                            if (!isOwner) {
+                                return failedFutureTxnNotOwned(txnID);
```

Comment:
It should not be the owner check, right? Essentially, it checks if the TC service is the super user.

Reply:
+1.
In this case, we need to change the name of the method and modify the failure message.

Reply:
And I have a question why we do not allow the transaction owner to end the transaction?
It is confused. A common user opens a transaction to do something, but he can not end the transaction since it's not a superuser.

Reply:
good point. 

the `endTxnOnPartition` and `endTxnOnSubscription` are targeted to the broker that owns the topic, so we might be able to read the tnxMeta (the related TC might be loaded in another broker) so the owner check is not doable. I think it's fine, my assumption is that the broker client auth MUST match a super-user.

Reply:
The `endTxnOnPartition` and `endTxnOnSubscription` are only called in `endTtansaction` which already has auth check, so I think the auth check for the superuser should be removed.

Reply:
A malicious attacker could still send this command impersonating the tc client, so we need to ensure that is allowed

Reply:
You are right, thanks for your answer.